commands/maliciousscan/maliciousscan.go

```diff
@@ -0,0 +1,319 @@
+	progress                  ioUtils.ProgressMgr
+	customAnalyzerManagerPath string
+}
+
+func (cmd *MaliciousScanCommand) SetProgress(progress ioUtils.ProgressMgr) {
+	cmd.progress = progress
+}
+
+func (cmd *MaliciousScanCommand) SetThreads(threads int) *MaliciousScanCommand {
+	cmd.threads = threads
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) SetServerDetails(server *config.ServerDetails) *MaliciousScanCommand {
+	cmd.serverDetails = server
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) SetWorkingDirs(workingDirs []string) *MaliciousScanCommand {
+	cmd.workingDirs = workingDirs
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) SetOutputFormat(format format.OutputFormat) *MaliciousScanCommand {
+	cmd.outputFormat = format
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) SetMinSeverityFilter(minSeverity severityutils.Severity) *MaliciousScanCommand {
+	cmd.minSeverityFilter = minSeverity
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) SetCustomAnalyzerManagerPath(path string) *MaliciousScanCommand {
+	cmd.customAnalyzerManagerPath = path
+	return cmd
+}
+
+func (cmd *MaliciousScanCommand) ServerDetails() (*config.ServerDetails, error) {
+	return cmd.serverDetails, nil
+}
+
+func (cmd *MaliciousScanCommand) CommandName() string {
+	return "malicious_scan"
+}
+
+func NewMaliciousScanCommand() *MaliciousScanCommand {
+	return &MaliciousScanCommand{}
+}
+
+func (cmd *MaliciousScanCommand) Run() (err error) {
+	defer func() {
+		if err != nil {
+			var e *exec.ExitError
+			if errors.As(err, &e) {
+				if e.ExitCode() != coreutils.ExitCodeVulnerableBuild.Code {
+					err = errors.New("Malicious scan command failed. " + err.Error())
+				}
+			}
+		}
+	}()
+
+	xrayVersion, entitledForJas, workingDirs, err := cmd.validateAndPrepare()
+	if err != nil {
+		return err
+	}
+
+	cmdResults := cmd.initializeCommandResults(xrayVersion, entitledForJas)
+	populateScanTargets(cmdResults, workingDirs)
+
+	scanner, err := cmd.createJasScanner()
+	if err != nil {
+		return err
+	}
+
+	if err = cmd.runMaliciousScans(cmdResults, scanner); err != nil {
+		return err
+	}
+
+	if cmd.progress != nil {
+		if err = cmd.progress.Quit(); err != nil {
+			return err
+		}
+	}
+
+	return cmd.outputResults(cmdResults)
+}
+
+func (cmd *MaliciousScanCommand) validateAndPrepare() (xrayVersion string, entitledForJas bool, workingDirs []string, err error) {
+	xrayManager, xrayVersion, err := xray.CreateXrayServiceManagerAndGetVersion(cmd.serverDetails)
+	if err != nil {
+		return "", false, nil, err
+	}
+
+	entitledForJas, err = jas.IsEntitledForJas(xrayManager, xrayVersion)
+	if err != nil {
+		return "", false, nil, err
+	}
+	if !entitledForJas {
+		return "", false, nil, errors.New("JAS (Advanced Security) feature is not entitled")
+	}
+
+	log.Info("JFrog Xray version is:", xrayVersion)
+
+	if err = jas.DownloadAnalyzerManagerIfNeeded(0); err != nil {
+		return "", false, nil, fmt.Errorf("failed to download Analyzer Manager: %w", err)
+	}
```

Comment:
```suggestion
    if cmd.customAnalyzerManagerPath == "" {
	    if err = jas.DownloadAnalyzerManagerIfNeeded(0); err != nil {
		    return "", false, nil, fmt.Errorf("failed to download Analyzer Manager: %w", err)
		}
	}
```
you are always download the AM, even if custom path is provided with the binary....

Reply:
I can see double call to `DownloadAnalyzerManagerIfNeeded`, maybe here is not needed?